react/components/ErrorPage/RefreshCounter.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { useEffect, useState } from 'react'
+import styles from './error-page.module.css'
+
+const EXPIRATION_TIME = 10 * 60 * 1000 // 10 minutes in milliseconds
+
+function RefreshCounter() {
+  const [counter, setCounter] = useState(10)
+
+  const hasRefreshed = localStorage.getItem('hasRefreshed')
+  const lastRefreshTime = (localStorage.getItem('lastRefreshTime') ||
+    0) as number
+  const currentTime = new Date().getTime()
+
+  const shouldRefresh =
+    !hasRefreshed ||
+    (lastRefreshTime && currentTime - lastRefreshTime > EXPIRATION_TIME)
+
+  useEffect(() => {
+    if (counter > 0) {
+      const timer = setInterval(() => setCounter(counter - 1), 1000)
+      return () => clearInterval(timer)
+    } else if (shouldRefresh) {
+      localStorage.setItem('hasRefreshed', 'true')
+      localStorage.setItem(
+        'lastRefreshTime',
+        (currentTime as unknown) as string
+      )
```

Comment:
```suggestion
  const currentTime = new Date().toISOString()

  const shouldRefresh =
    !hasRefreshed ||
    (lastRefreshTime && currentTime - lastRefreshTime > EXPIRATION_TIME)

  useEffect(() => {
    if (counter > 0) {
      const timer = setInterval(() => setCounter(counter - 1), 1000)
      return () => clearInterval(timer)
    } else if (shouldRefresh) {
      localStorage.setItem('hasRefreshed', 'true')
      localStorage.setItem('lastRefreshTime', currentTime)
```

Reply:
Tested your suggestion, it didn't work. Because I need the date to be a number for this operation

` const shouldRefresh =
    !hasRefreshed ||
    (lastRefreshTime && currentTime - lastRefreshTime > EXPIRATION_TIME)` 
    
    using `new Date().toISOString()` didn't work for that operation, not even forcing it to be a number (forcing returns NaN)